src/Stack/Constants.hs

```diff
@@ -216,3 +218,9 @@ compilerOptionsCabalFlag Ghcjs = "--ghcjs-options"
 
 ghcColorForceFlag :: String
 ghcColorForceFlag = "-fdiagnostics-color=always"
+
+minTerminalWidth :: Int
+minTerminalWidth = 20
```

Comment:
20 is **really** small.  I think 40 might be a better minimum.  Reasoning is that the user might resize their terminal.

I think having a max makes sense too, it's unpleasant to read a whole paragraph of text stretched on a line.  Maybe 200 chars?  Of course, an answer to that would be why doesn't the user resize their terminal.  I use XMonad, though, so if I open up a terminal on a 4k display, I probably don't really want paragraphs to be single lines, though that does happen.

Reply:
Yeah I wasn't sure what to do for a min. 40 does look more reasonable. I'm indifferent to a maximum personally, but that reasoning seems to make sense, I'll throw it in.

Hm, I just noticed that I'm doing the checking all wrong too, it's only checking the user-supplied one, and it's just ignoring it completely if it's too small. Should probably just clip to the allowed range if it's outside it. I'll redo that part, heh.